static/js/controllers/contacts-content.js

```diff
@@ -218,12 +211,28 @@ var _ = require('underscore'),
         });
       };
 
-      var sortChildPersons = function(persons, contactDoc) {
-        if (persons && persons.length) {
+      var removePrimaryContact = function(children, primaryContactDoc) {
+        if (!children || children.length === 0 || !primaryContactDoc) {
+          return;
+        }
+        var primaryContactIdx = _.findIndex(children, function(child) {
+          return child.doc._id === primaryContactDoc._id;
+        });
+        if (primaryContactIdx < 0) {
+          return;
+        }
+        children.splice(primaryContactIdx, 1);
+      };
+
+      var sortChildPersons = function(persons, contactDoc, primaryContactDoc) {
+        if (persons.length) {
           if (contactDoc.type === 'clinic') {
             sortFamilyMembers(persons);
           }
-          shiftAndMarkContactPerson(persons, contactDoc);
+        }
+        if (primaryContactDoc) {
+          removePrimaryContact(persons, primaryContactDoc);
+          persons.unshift({ id: primaryContactDoc._id, doc: primaryContactDoc, isPrimaryContact: true });
```

Comment:
So this `if` block is basically moving the primary contact (if there is one) to the front of `persons`? It would be clearer if there was a function that performed that (ie `removePrimaryContact` could be a sub call, but the one thing you're trying to achieve is moving the contact to the front).